File: android/src/main/java/com/example/app_preferences_flutter/PreferencesActivity.java
package com.example.app_preferences_flutter;
import android.os.Bundle;
import android.preference.PreferenceActivity;
import android.preference.PreferenceFragment;

public class PreferencesActivity  extends PreferenceActivity {
    @Override
    public void onCreate(Bundle savedInstanceState)  {
        super.onCreate(savedInstanceState);

        // load settings fragment
        getFragmentManager().beginTransaction().replace(android.R.id.content, new MainPreferenceFragment()).commit();
    }

    public static class MainPreferenceFragment extends PreferenceFragment {
        @Override
        public void onCreate(final Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            addPreferencesFromResource(R.xml.preferences);
        }
    }
}

File: android/src/main/java/com/danielisho/app_preferences_flutter/PreferencesActivity.kt
package com.danielisho.app_preferences_flutter

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity


class PreferencesActivity: AppCompatActivity(){

        override fun onCreate(savedInstanceState: Bundle?) {
                super.onCreate(savedInstanceState)
                setContentView(R.layout.activity_main)

                supportActionBar!!.setDisplayHomeAsUpEnabled(true)
                supportActionBar!!.setHomeButtonEnabled(true)

                supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.settings, PreferencesFragment())
                        .commit()
        }

        override fun onSupportNavigateUp(): Boolean {
                onBackPressed()
                return true
        }
}